version/src/lib.rs

```diff
@@ -1,59 +1,50 @@
 #![cfg_attr(RUSTC_WITH_SPECIALIZATION, feature(min_specialization))]
 
 extern crate serde_derive;
+pub use self::legacy::{LegacyVersion1, LegacyVersion2};
 use {
     serde_derive::{Deserialize, Serialize},
-    solana_sdk::sanitize::Sanitize,
+    solana_sdk::{sanitize::Sanitize, serde_varint},
     std::{convert::TryInto, fmt},
 };
 #[macro_use]
 extern crate solana_frozen_abi_macro;
 
-// Older version structure used earlier 1.3.x releases
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, AbiExample)]
-pub struct LegacyVersion {
-    major: u16,
-    minor: u16,
-    patch: u16,
-    commit: Option<u32>, // first 4 bytes of the sha1 commit hash
-}
+mod legacy;
 
-impl Sanitize for LegacyVersion {}
+#[derive(Debug)]
+enum ClientId {
+    SolanaLabs,
```

Comment:
I feel like we can pre-allocate Jito and Firedancer here too since they're already known clients

Reply:
done.

Reply:
@buffalu - 👀, please let us know if this doesn't look good for you guys

Reply:
lgtm

Reply:
if/when we deprecate jito as plugin, how does that change this client id? should we put firedancer before jito?

Reply:
client id just gets burned in that case. nbd